sympy/solvers/tests/test_solvers.py

```diff
@@ -1997,3 +1994,33 @@ def test_issue_15415():
     assert solve(Eq(y + 3*x**2/2, y + 3*x), y) == []
     assert solve([Eq(y + 3*x**2/2, y + 3*x)], y) == []
     assert solve([Eq(y + 3*x**2/2, y + 3*x), Eq(x, 1)], y) == []
+
+def test_issue_15731():
+    # f(x)**g(x)=c
+    assert solve(Eq((x**2 - 7*x + 11)**(x**2 - 13*x + 42), 1)) == [2, 3, 4, 5, 6, 7]
+    assert solve((x)**(x + 4) - 4) == [-2]
+    assert solve((-x)**(-x + 4) - 4) == [2]
+    assert solve((x**2 - 6)**(x**2 - 2) - 4) == [-2, 2]
+    assert solve((x**2 - 2*x - 1)**(x**2 - 3) - 1/(1 - 2*sqrt(2))) == [sqrt(2)]
+    assert solve(x**(x + 1/2) - 4*sqrt(2)) == [S(2)]
```

Comment:
Hmm...is this the same as `solve(x**(x + 0.5) - 4*sqrt(2)) == [S(2)]`? If so, it shouldn't be because the 0.5 is inexact while the solution is exact. If this comes from nsimplify giving an integer for a decimal input then we should be careful to not replace the decimal answer with rational one.

Reply:
No actually the solutions are like this when you run this test case
```
solve(x**(x + 0.5) - 4*sqrt(2))
[2.00000000000000]

solve(x**(x + S(1)/2) - 4*sqrt(2))
[2]
```
I will change the test case.